src/frontend/src/optimizer/plan_node/stream_table_scan.rs

```diff
@@ -42,8 +42,8 @@ impl StreamTableScan {
             ctx,
             logical.schema().clone(),
             logical.base.pk_indices.clone(),
-            Distribution::AnyShard,
-            false, // TODO: determine the `append-only` field of table scan
+            Distribution::AnyShard, // Mark as `AnyShard` cause we don't know the distribution yet.
```

Comment:
The distribution of a table should be stored in catalog. May resolve this issue later.

Reply:
Will fix this in a separate PR.